File: src/main/java/com/vaadin/demo/component/menubar/MenuBarRightAligned.java
package com.vaadin.demo.component.menubar;

import com.vaadin.flow.component.contextmenu.MenuItem;
import com.vaadin.flow.component.contextmenu.SubMenu;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.menubar.MenuBar;
import com.vaadin.flow.component.menubar.MenuBarVariant;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("menu-bar-right-aligned")
public class MenuBarRightAligned extends Div {

    public MenuBarRightAligned() {
        // tag::snippet[]
        MenuBar menuBar = new MenuBar();
        menuBar.addThemeVariants(MenuBarVariant.LUMO_END_ALIGNED);
        // end::snippet[]

        menuBar.addItem("View");
        menuBar.addItem("Edit");
        MenuItem share = menuBar.addItem("Share");
        SubMenu shareSubMenu = share.getSubMenu();
        shareSubMenu.addItem("By email");
        shareSubMenu.addItem("Get Link");

        add(menuBar);
    }

    public static class Exporter extends DemoExporter<MenuBarRightAligned> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/menubar/menu-bar-right-aligned.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import '@vaadin/menu-bar';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('menu-bar-right-aligned')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  @state()
  private items = [
    { text: 'View' },
    { text: 'Edit' },
    {
      text: 'Share',
      children: [{ text: 'By email' }, { text: 'Get link' }],
    },
  ];

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-menu-bar theme="end-aligned" .items="${this.items}"></vaadin-menu-bar>
      <!-- end::snippet[] -->
    `;
  }
}
